### PR TITLE
On pan drag

### PR DESCRIPTION
**Pull request by jrichardlai**
Allow pan drag event from the map, this allow to drawing more smoothly polyline and polygons.

iOS seems to be fine, but Android seems a little laggy on Simulator and would crash after lots of drawings, ( which has not happened yet on the phone for me ), but I'd assume its about the high number of `setState` calls.

Note: for iOS `scrollEnabled` should be set to false to trigger the event

### CODE REVIEW

**Comment by lelandrichardson** [comment]
@jrichardlai do you have a gif of this in action? Pretty cool.

**Comment by jrichardlai** [comment]
Hey @lelandrichardson, sorry been pretty busy lately, here is a gif. In Android it's pretty laggy and would crash if you do a setState for all coordinates, so I'm adding a point for our app only after a delta.
We've been using it in prod for a week with no problems so far. Do you know a better way to send the drag with Android.

out

**Comment by christopherdro** [comment]
@jrichardlai LGTM! How has this been performing in the last week. Still good?

/cc @lelandrichardson Can you look over this and approve?

**Comment by lelandrichardson** [comment]
@jrichardlai @christopherdro this looks good to me. It's correct to assume that there is no performance degradation if you aren't using this event, right?

**Comment by christopherdro** [comment]
@jrichardlai Could we also add 'Note: for iOS scrollEnabled should be set to false to trigger the event' inside the docs somewhere?

**Comment by jrichardlai** [comment]
@lelandrichardson @christopherdro yes no issues noted so far.

For the performance in iOS I am still adding a `GestureResponder` so `handleMapDrag` will still be called but will return if `onPanDrag` is not set.
However on Android, its using `getExportedCustomDirectEventTypeConstants` and so `onPanDrag` is called on the drag, we didn't noticed an issue with scrolling but I think the performance could be improved, is there a way to return like with iOS?
I'll update the doc to have a note for the iOS scrollEnabled.

**Comment by lelandrichardson** [comment]
hmm. not sure the right way to solve that...

**Comment by lelandrichardson** [comment]
:shipit:

**Comment by jrichardlai** [comment]
@lelandrichardson I'm gonna do more testing for performance on Android

**Comment by jrichardlai** [comment]
@lelandrichardson after some testing on a less powerful phone I noticed that there was unfortunately some lag issues because of the onPanDrag call. I made a little tweak https://github.com/lelandrichardson/react-native-maps/pull/159/commits/6253f7bee1778dee6bffbd4a4472c0dcd3c96bed is there another way you think we can fix that?

**Comment by jrichardlai** [comment]
@lelandrichardson I think this PR is gtg been using this code for few weeks now

File: android/lib/src/main/java/com/airbnb/android/react/maps/AirMapView.java [hunk 1]
```diff
@@ -80,12 +82,12 @@ public AirMapView(ThemedReactContext context, AirMapManager manager) {
 //                return false;
 //            }
 
-                    @Override
-                    public boolean onScaleBegin(ScaleGestureDetector detector) {
-                        view.startMonitoringRegion();
-                        return true; // stop recording this gesture. let mapview handle it.
-                    }
-                });
```
Comment (jrichardlai) [inline android/lib/src/main/java/com/airbnb/android/react/maps/AirMapView.java]: I'll fix the spaces

**Comment by jrichardlai** [comment]
Thanks for merging @christopherdro ! 👍

**Comment by yaronlevi** [comment]
Hi can you please reference a place in the docs where it explains about OnPanDrag or how to achieve the line draw you built in the GIF above?

I've looked all over the docs and can not find this.
Any help will be much appreciated.

**Comment by jrichardlai** [comment]
You can look at the example https://github.com/airbnb/react-native-maps/blob/master/example/examples/PolylineCreator.js let me know if that helps

**Comment by yaronlevi** [comment]
@jrichardlai Thank! just what I was looking for. Going to try it out.

**Comment by cgelosulit** [comment]
"Was this removed in the example?"

**Pull request merged**